public/js/clients/firefox/events.js

```diff
@@ -59,19 +65,18 @@ function resumed(_, packet) {
 function newSource(_, packet) {
   const { source } = packet;
 
-  if (NEW_SOURCE_IGNORED_URLS.indexOf(source.url) > -1) {
-    return;
-  }
-
-  if (source.introductionType == "debugger eval") {
-    return;
+  source.hidden = false;
+  if (isHidden(source)) {
+    source.url = `evals/console-evauation-${evalIndex++}.js`;
+    source.hidden = true;
```

Comment:
Thanks for the detailed answer, this helps provide some clarity.

I have a test for "break on next", where the debuggee pauses on a debugger eval. Another example would be a console evaluation that has a debugger statement in it. In both cases, the debuggee is paused and the debugger wants to show the debugger eval source, source tab, and call stack. 

> Your changes here would ignore valid eval-based sources that are given a URL with the sourceURL pragma

I have a new `evaluations` test that hopefully will catch these changes as well so we can lock down the right behavior.